### PR TITLE
Adding telemetry for installing tools

### PR DESCRIPTION
**Pull request by kiiadi**

### CODE REVIEW

File: jetbrains-core/resources/telemetryOverride.json [hunk 1]
```diff
@@ -10,6 +10,12 @@
             "name": "experimentId",
             "type": "string",
             "description": "The id of the experiment being activated or deactivated"
+        },
+        {
+            "name": "toolId",
+            "type": "string",
+            "description": "The tool being installed",
+            "allowedValues": ["ssm-plugin", "sam-cli"]
```
Comment (rli) [inline jetbrains-core/resources/telemetryOverride.json:18]: nit: `session-manager-plugin` and `aws-sam-cli`?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/DefaultToolManager.kt [hunk 2]
```diff
@@ -181,13 +183,13 @@ class DefaultToolManager @NonInjectable internal constructor(private val clock:
...
                 override fun run(indicator: ProgressIndicator) {
-                    installTool(type, indicator)
+                    installTool(type, indicator, project)
                 }
             }
         )
     }
 
-    private fun <V : Version> installTool(type: ManagedToolType<V>, indicator: ProgressIndicator?): Tool<ToolType<V>> {
+    private fun <V : Version> installTool(type: ManagedToolType<V>, indicator: ProgressIndicator?, project: Project?): Tool<ToolType<V>> {
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/DefaultToolManager.kt:192]: nit: i feel like typically we try to put project first

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/ToolType.kt [hunk 3]
```diff
@@ -47,6 +48,13 @@ interface AutoDetectableToolType<VersionScheme : Version> : ToolType<VersionSche
 }
 
 interface ManagedToolType<VersionScheme : Version> : ToolType<VersionScheme> {
+    override val id: String
+        get() = telemetryId.toString()
+
+    /**
+     * Used to identify this tool with the telemetry stack
+     */
+    val telemetryId: ToolId
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/ToolType.kt:57]: is it even worth differentiating between the two?

File: jetbrains-core/resources/telemetryOverride.json [hunk 4]
```diff
@@ -33,6 +39,18 @@
                     "type": "experimentState"
                 }
             ]
+        },
+        {
+            "name": "aws_toolInstallation",
```
Comment (rli) [inline jetbrains-core/resources/telemetryOverride.json:44]: future-proof by adding an automatic vs manual invocation?

**Review by rli — APPROVED** [review]

File: jetbrains-core/resources/telemetryOverride.json (hunk 1 above)
Comment (kiiadi) [reply jetbrains-core/resources/telemetryOverride.json:18]: +1

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/ToolType.kt (hunk 3 above)
Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/ToolType.kt:57]: possibly not but I didn't want to leak the `ToolId` enum all the way through given most of the time all we need is a String ...

File: jetbrains-core/resources/telemetryOverride.json (hunk 4 above)
Comment (kiiadi) [reply jetbrains-core/resources/telemetryOverride.json:44]: what does that mean?

Comment (rli) [reply jetbrains-core/resources/telemetryOverride.json:44]: we don't expose it now, but i assumed eventually we'd want to allow users to manually trigger install/update. can definitely tackle the telemetry definition for that when it's actually implemented

Comment (JadenSimon) [reply jetbrains-core/resources/telemetryOverride.json:44]: I guess add two metadata flags: `automatic` and `updated`.
Though `passive` covers the manual vs. automatic right?

Comment (kiiadi) [reply jetbrains-core/resources/telemetryOverride.json:44]: the primary question I'm trying to answer here - is how many automated installations fail - this is something we can (potentially fix)

Comment (rli) [reply jetbrains-core/resources/telemetryOverride.json:44]: okay, we can keep as-is for now then

**Pull request merged**